tests/test_models.py

```diff
@@ -0,0 +1,52 @@
+"""Tests for the Post model."""
+
+from datetime import datetime
+from unittest.mock import patch
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.utils import timezone
+from blog.models import Post
+
+User = get_user_model()
+
+
+class TestPostModel(TestCase):
+    """Tests for the Post model."""
+
+    def setUp(self):
+        """"Create test User and Post objects."""
+        self.user = User.objects.create(username='test user')
+        self.test_post = Post.objects.create(
+            author=self.user,
+            title='test title',
+            text='test text',
+        )
+
+    @patch(
+        'django.utils.timezone.now', lambda: datetime(
+            day=12,
+            month=1,
+            year=2024,
+            tzinfo=timezone.get_current_timezone(),
+        ),
+    )
+    def test_post_publish(self):
+        """Ensure correct datetime for the published posts."""
```

Comment:
I'd say that this should check that it changed from `None` to the target date.
```suggestion
        """Ensure publishing a post sets its date."""
```